src/main/java/seedu/address/model/statistics/Bill.java

```diff
@@ -4,63 +4,87 @@
 
 import java.util.Objects;
 
+import javafx.collections.ObservableList;
+import seedu.address.model.menu.Menu;
+import seedu.address.model.menu.MenuItem;
+import seedu.address.model.order.OrderItem;
+import seedu.address.model.order.UniqueOrderItemList;
+import seedu.address.model.table.Table;
+import seedu.address.model.table.TableNumber;
+
 /**
  * Represents a Bill from a UniqueOrderItemList from the RestOrRant.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
 public class Bill {
 
-    private final int tableNumber;
-    private final int day;
-    private final int month;
-    private final int year;
+    private final Table table;
+    private final TableNumber tableNumber;
+    private final Day day;
+    private final Month month;
+    private final Year year;
     private final int totalBill;
-    //private final Table table;
 
     /**
      * Every field must be present and not null.
      */
-    public Bill(int tableNumber, int day, int month, int year) {
-        requireAllNonNull(tableNumber);
-        this.tableNumber = tableNumber;
-        //this.table = getTable(tableNumber);
+    public Bill(Table table, Day day, Month month, Year year) {
+        requireAllNonNull(table);
+        requireAllNonNull(day);
+        requireAllNonNull(month);
+        requireAllNonNull(year);
+        this.table = table;
+        this.tableNumber = table.getTableNumber();
         this.day = day;
         this.month = month;
         this.year = year;
-        this.totalBill = calculateBill();
+        this.totalBill = 0;
     }
 
-    private int calculateBill() {
-        int totalBill = 0;
+    public void calculateBill() {
         // TODO: iterate through the UniqueOrderList from the Table
         // Use getMenuItem().getPrice() after merging with the rest
-        return totalBill;
+        //for each orderItem in the UniqueOrderItemList
+        //if the table
+        Menu menu = new Menu();
+        MenuItem menuItem;
+        ObservableList<OrderItem> orderItemList = table.getUniqueOrderItemList().asUnmodifiableObservableList();
```

Comment:
You should get the filteredOrderItemList from ModelManager!

Reply:
Ok, done! Thank you!